ext/intl/common/common_date.cpp

```diff
@@ -137,7 +139,8 @@ U_CFUNC int intl_datetime_decompose(zval *z, double *millis, TimeZone **tz,
 			return FAILURE;
 		}
 
-		*millis = U_MILLIS_PER_SECOND * (double)Z_LVAL(retval);
+		datetime = Z_PHPDATE_P(z);
+		*millis = U_MILLIS_PER_SECOND * ((double)Z_LVAL(retval) + datetime->time->f);
```

Comment:
The fact that the timestamp is retrieved through a `getTimestamp()` method while the fraction is obtained directly from the structure feels like a breach of abstraction. On the other hand, it does not seem like `DateTime` has a direct way of getting the fraction (or timestamp with fraction).

@derickr Do you have any thoughts on this?